src/microbe_stage/editor/CellEditorComponent.GUI.cs

```diff
@@ -302,14 +312,101 @@ private void UpdateOrganelleUnlockTooltips()
+            var control = entry.Value;
+
+            // Skip already unlocked organelles
+            if (Editor.CurrentGame.GameWorld.UnlockProgress.IsUnlocked(organelle, Editor.CurrentGame, autoUnlock))
+            {
+                control.Undiscovered = false;
+                continue;
+            }
+
+            // Skip hidden organelles unless they are hidden because of missing requirements
+            if (!control.Visible && !control.Undiscovered)
+                continue;
+
+            control.Hide();
+            control.Undiscovered = true;
+
+            var buttonGroup = organelle.EditorButtonGroup;
+            var unlockTextString = new LocalizedString("UNLOCK_WITH_EITHER_OF_FOLLOWING", organelle.Name);
+
+            // Create unlock text
+            if (groupsWithUndiscoveredOrganelles.TryGetValue(buttonGroup, out var group))
+            {
+                // Add a new organelle to the group
+                group.Count += 1;
+                group.UnlockText.Append("\n\n");
+                group.UnlockText.Append(unlockTextString);
+                group.UnlockText.Append(" ");
+                organelle.GenerateUnlockRequirementsText(group.UnlockText);
+                groupsWithUndiscoveredOrganelles[buttonGroup] = group;
+            }
+            else
+            {
+                // Add the first organelle to the group
+                var unlockText = new LocalizedStringBuilder();
+                unlockText.Append(unlockTextString);
+                unlockText.Append(" ");
+                organelle.GenerateUnlockRequirementsText(unlockText);
+                groupsWithUndiscoveredOrganelles.Add(buttonGroup, (unlockText, 1));
+            }
+        }
+
+        // Remove any buttons that might've been created before
+        if (refresh)
+            RemoveUndisoveredOrganelleButtons();
+
+        // Generate undiscovered organelle buttons
+        foreach (var groupWithUndiscovered in groupsWithUndiscoveredOrganelles)
+        {
+            var group = partsSelectionContainer.GetNode<CollapsibleList>(groupWithUndiscovered.Key.ToString());
+            var (unlockText, count) = groupWithUndiscovered.Value;
+
+            var button = (UndiscoveredOrganellesButton)undiscoveredOrganellesScene.Instance();
+            button.Count = count;
+            group.AddItem(button);
+
+            // Register tooltip
+            var tooltip = (UndiscoveredOrganellesTooltip)undiscoveredOrganellesTooltipScene.Instance();
+            tooltip.UnlockText = unlockText;
+            ToolTipManager.Instance.AddToolTip(tooltip, "lockedOrganelles");
+            button.RegisterToolTipForControl(tooltip, true);
+        }
+    }
+
+    private void RemoveUndisoveredOrganelleButtons()
+    {
+        foreach (var child in partsSelectionContainer.GetChildren())
+        {
+            if (child is CollapsibleList list)
+                list.RemoveAllOfType<UndiscoveredOrganellesButton>();
```

Comment:
Does this need to unregister the tooltips? I can't immediately remember if the AddToolTip call above works so that it automatically unregisters the tooltips.

I think as the specific group is used, it would be pretty simple to clear that group here.

Reply:
Well it does have `autoUnregister` set to true so it should work

Reply:
That's just the tooltip control data, right? I've never been 100% familiar with the tooltip system, but there's two different things: the core tooltip data, and then the control <-> tooltip mapping (this second type of data is the one with the auto unregister property).

Reply:
I'm unresolving this as I don't see this code changed (sorry if you addressed this in one of your commits, but it's been way too much for me to read through each one).

Reply:
Pretty sure the call to `ToolTipManager.Instance.ClearToolTips` under this loop deletes all the related tooltips in the group.

Reply:
Yep, I see it now. I had once again accidentally been reading the outdated diff on github (as it is not very visible the one button you need to press to view all changes).